client/src/firebaseInit.js

```diff
@@ -0,0 +1,56 @@
+import firebase from "firebase/compat/app";
+import { getMessaging, getToken, onMessage } from "firebase/messaging";
+import { toast } from "react-hot-toast";
+
+const config = {
+  apiKey: "AIzaSyAgUY5d0kc556sShMaMogBmfcLRNZ_ClDU",
+  authDomain: "studydash-fd048.firebaseapp.com",
+  projectId: "studydash-fd048",
+  storageBucket: "studydash-fd048.appspot.com",
+  messagingSenderId: "66459387705",
+  appId: "1:66459387705:web:8e01b1649d8fd481dd8ae8",
+  measurementId: "G-ELZW5L303E",
+};
+
+const app = firebase.initializeApp(config);
+const messaging = getMessaging(app);
+
+// next block of code goes here
+export const fetchToken = async (setToken) => {
+  await Notification.requestPermission().then((permission) => {
```

Comment:
Perhaps we should allow users to use the application without granting notification permissions. Could be a deal breaker for users who do not approve of the setting being forced on them.

Reply:
You seem to be able to disable the notification permission after logging in anyways.

Reply:
True true, I just did that to ensure notifs were working at that point, but I can definitely shift this to the chat page and ask for permissions there before users continue using that functionality...

Reply:
I feel like some users may just want no notifications at all (even in chat) so maybe don't make it mandatory there (if that was what you were planning on doing)

Reply:
or just tweak it a bit to work normally without notifs